### DISCUSSION TITLE
Keep ordering for ExcelUnknownCells

### DISCUSSION THREAD

**Discussion by muraval** [ozlerhakan/poiji#313] [Ideas]
Hey, @ozlerhakan! Thanks for incredible amount of work you did for this miraculous project.

Currently I work with files that don't have a defined naming and amount of columns. So I had to stick to @ExcelUnknownCells annotation. 

Is it possible to keep the order of headers the way how they appeared in original file instead of storing them in HashMap where they get hashed and reordered? As a solution, for example, can you use a LinkedHashMap?

Thanks!

**Comment by ozlerhakan** [comment]
Hi @muraval , I'd say that makes sense to me! Do you want to contribute Poiji, which I would be glad?

**Comment by vaa25** [comment]
@muraval  Hi! you can try https://github.com/vaa25/poiji2 . It keeps ordering for ExcelUnknownCells.

**Reply by fatihyildizhan** [reply]
@vaa25 Hello, I've checked at the poiji2 project and noticed a situation that I was very shocked. You committed the codes written by @ozlerhakan, who developed the poiji project, as if you wrote them. Then you are trying to direct us to this project. 

If you want to write a project like poiji, write it. No one interferes with this. Or you can fork poiji and keep adding new features. However, you commit the codes written by @ozlerhakan as if you wrote them. 

Why are you doing this? This is VERY unethical. Please stop this!

**Reply by vaa25** [reply]
@fatihyildizhan Thank you for your feedback. I have made some improvements to poiji some years ago and waited months  until @ozlerhakan check my pull request. I had no time to wait more, but I needed new functionality. So I created new repository with improvements. Just I closed pull requests @ozlerhakan has appeared, but it was too late. @ozlerhakan has created great project with excellent github face, but as I see he still ignores consumers. I try to fix this problem in my project. Now I see people needs better excel tool. So I try to provide what they need. 
> you commit the codes written by @ozlerhakan as if you wrote them.

How can you say these words? First words of poiji2 description are: "This is an advanced fork of https://github.com/ozlerhakan/poiji." 
Poiji2 has many old code. I didn't change author's comments to keep respect for him. But if you demand, I'll rework classes and remove every Hakan and other mentions, Also I'll remove link to original poiji, if @ozlerhakan asks. He never tried to contact me since that time, by the way.

If you have any questions, don't hesitate to ask, please. But I am not going to stop developing poiji2. When it growth more and become more different from original poiji, I'll make rebranding.

**Comment by ozlerhakan** [comment]
Ok, this is my first and last naive comment that will be concise and direct about this annoying situation where you (@vaa25 ) have insisted to build a project on top of the history implemented in Poiji with the help of all the other contributors here (@breucode , @aerfus , @brunslo , @wilds , @Brutus5000 , @ghacupha , @itsjwala , @beckerdennis , @kpolli , @thiyagu-7 , @bendh , @jrse , @abdulrafique , @virtual-machinist , @4sujittiwari , @jmorgan-iad , @fatihyildizhan , @soberich , and @chinmay1994 ) with a new name so-called "poiji2", which ENTIRELY against the license! 

From day one until today, Poiji has always aimed to solve one problem: a means to convert Excel data to a list of Java items! However, I have been constantly mindful over the years and respected others' thoughts when they come up with new ideas, including you as well! Poiji has been a community-driven project, evolving with the open-source community.

> I have made some improvements to poiji some years ago and waited months until @ozlerhakan check my pull request. I had no time to wait more, but I needed new functionality. So I created new repository with improvements. Just I closed pull requests @ozlerhakan has appeared, but it was too late.

Sorry, but I have other responsibilities besides this project, and this statement is truly subjective. See https://github.com/ozlerhakan/poiji/pull/129#issuecomment-584304105.

> but as I see he still ignores consumers.

I have always encouraged people like you See https://github.com/ozlerhakan/poiji/issues/124#issuecomment-575980263. At the time, there were other maintainers that attempted to introduce Java -> Excel functionality  however, the maintainer didn't complete the task See https://github.com/ozlerhakan/poiji/pull/235. Plus, over 7 years, around 79 PRs have been merged (excluding releases), while there are less than 10 closed PRs. So who is actually against contribution?


> Now I see people needs better excel tool.

There can be new requirements and use cases based on different experiences. There is no one-size-fits-all project in the world. But from your perspective time is up for you to wait and you ended up copying the implementation from the original source without crediting authors :) 

>  But if you demand, I'll rework classes and remove every Hakan and other mentions, Also I'll remove link to original poiji.

So let me understand, you will erase the entire history by copy-pasting all the implementations and rebrand it with a new project. Haha. This is a clever idea :D.

> He never tried to contact me since that time, by the way.

That's totally a desperate speech. I approached you kindly, even though you never got back to me. (SEE https://github.com/ozlerhakan/poiji/issues/171#issuecomment-727251699).


> But I am not going to stop developing poiji2. When it growth more and become more different from original poiji, I'll make rebranding.

Remember that you're still exploiting Poiji's community to promote your project!! As of today, whenever you comment on either a discussion or issue related to your project, it will be either deleted or edited by me. Apparently, we don't talk the same way, so I'd say this is the only alternative you will likely understand.

Best,
Hakan

**Reply by Brutus5000** [reply]
@vaa25 Just to clarify one thing: You can build your own fork, rename it (even though taking the same name and just bumping the number leaves a bad taste) and do whatever you want. The license is very free. The only thing you have to do is keep the git history so all of us authors are still named. There is absolutely no reason not do this. You don't lose anything.

The good thing is: you can still fix this, if you rebase your branch against whatever commit you originally started with, you can recover the history and then force push on your branch. And then you can go nuts on your project.

**Reply by vaa25** [reply]
@ozlerhakan oh, I don't remember https://github.com/ozlerhakan/poiji/issues/171#issuecomment-727251699 . Probably I didn't see it. Sure you may edit or delete my messages. It is ok.
@Brutus5000 and others. Sorry, guys, I didn't want to abuse you all. How can I fix situation with contributor mentioning? Technically poiji2 is not a fork, and I don't remember why I didn't develop forked repository (probably there were some problems with publishing). I doubt it is possible to tie it to poiji for rebasing. Is it ok to list contributors in readme? Or you have other option?  Help me do good things right way, please.

**Reply by Brutus5000** [reply]
@vaa25 I did not have time to look into why rebasing would. Another simple approach would be: Checkout the history of this repo at a commit close to where you started forking (bring it in from a remote branch if required). Use this as a starting point for your branch. Then copy over the content of the first commit of your current fork repo. Then "cherry pick" the whole of your commits (except the first of course) on top of it.
